dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/storage/S3BackedImageTagStore.java

```
@@ -0,0 +1,121 @@
+            String key = os.getKey();
+            imageNameWithAccessTime.put(key, lastModified);
+        }
+        imageNamesWithTag = getStoreContentSortedByAccessDate(imageNameWithAccessTime);
+        return imageNamesWithTag;
+    }
+
+    private List<ImageTagStoreContent> getStoreContentSortedByAccessDate(Map<String, Date> imageNameWithAccessTime) throws InterruptedException {
+        List<ImageTagStoreContent> imageNameWithTagSortedByAccessDate = new ArrayList<>();
+        LinkedHashMap<String, Date> sortedResult = new LinkedHashMap<>();
+        // Sort the content by the access date so that the file which was accessed most recently gets processed first
+        imageNameWithAccessTime.entrySet()
+                .stream()
+                .sorted(Map.Entry.comparingByValue(Comparator.reverseOrder()))
+                .forEachOrdered(x -> sortedResult.put(x.getKey(), x.getValue()));
+
+        for (Map.Entry<String, Date> set : sortedResult.entrySet()) {
+            String key = set.getKey();
+            try {
+                S3Object o = getS3Object(store, key);
+                String image = convertS3ObjectKeyToImageString(key);
+                String tag = getTagValueFromObject(o);
+                ImageTagStoreContent imageTagStoreContent = new ImageTagStoreContent(image, tag);
+                imageNameWithTagSortedByAccessDate.add(imageTagStoreContent);
+            } catch (Exception e){
+                log.error("Encountered issues reading S3 object with key {}. Exception: {}", key, e);
+            }
+        }
+        return imageNameWithTagSortedByAccessDate;
+    }
+
+    private String getTagValueFromObject(S3Object o) {
+        String tag = "";
+        try (S3ObjectInputStream is = o.getObjectContent()) {
+            tag = CharStreams.toString(new InputStreamReader(is, Charsets.UTF_8));
+        } catch (IOException e) {
+            log.error("Could not get the tag value. Exception: ", e.getMessage());
```

Is there is a reason we are not bubbling this exception up and swallowing it here instead?